src/dashboard/src/components/ingest/views.py

```diff
@@ -425,6 +425,79 @@ def _es_results_to_directory_tree(path, return_list, not_draggable=False):
+        This must be in the new format defined for Archivematica 1.6.
+    :param dict record_map: A dictionary to be used to track created directory objects.
+        This ensures that duplicate directories aren't created when processing multiple files.
+    :param dict directory_list: A list of top-level directories to return in the results.
+        This only contains directories which are not themselves contained within any other directories, e.g. transfers.
+        This will be appended to by this function in lieu of returning a value.
+    :param bool not_draggable: This property determines whether or not a given file should be able to be dragged in the user interface; passing this will override the default logic for determining if a file is draggable.
+    """
+    dir, fn = record['relative_path'].rsplit('/', 1)
+
+    # Recursively create elements for this item's parent directory,
+    # if not already present in the record map
+    components = dir.split('/')
+    directories = []
+    while len(components) > 0:
+        directories.insert(0, '/'.join(components))
+        components.pop(-1)
+
+    parent = None
+    for node in directories:
+        is_transfer = len(node.split('/')) == 1
+
+        if not node in record_map:
+            dir_record = {
+                'type': 'transfer' if is_transfer else 'directory',
+                # have to artificially create directory IDs, since we don't assign those
+                'id': str(uuid.uuid4()),
+                'title': base64.b64encode(os.path.basename(node)),
+                'relative_path': base64.b64encode(node),
+                'not_draggable': not_draggable,
+                'object_count': 0,
+                'children': [],
+            }
+            record_map[node] = dir_record
+            # directory_list should consist only of top-level records
+            if is_transfer:
+                directory_list.append(dir_record)
+        else:
+            dir_record = record_map[node]
+
+        if parent is not None and not dir_record in parent['children']:
+            parent['children'].append(dir_record)
+            object_count = parent['object_count'] + 1
+            parent['object_count'] = object_count
+
+        parent = dir_record
+
+    child = {
+        'type': 'file',
+        'id': record['fileuuid'],
+        'title': base64.b64encode(fn),
+        'relative_path': base64.b64encode(record['relative_path']),
+        'size': record['size'],
+        'tags': record['tags'],
+        'bulk_extractor_reports': record['bulk_extractor_reports'],
+        'not_draggable': not_draggable
+    }
+    if record['format']:
+        format = record['format'][0]  # TODO handle multiple format identifications
+        child['format'] = format['format']
+        child['group'] = format['group']
+        child['puid'] = format['puid']
+
+    record_map[dir]['children'].append(child)
+    record_map[dir]['object_count'] += 1
```

Comment:
Is `object_count` supposed to be the count of children or the count of all descendants?